Actors docs refresh (#3300)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/actors/actors-overview.md

While your code processes a message, it can send one or more messages to other actors, or create new actors. An underlying runtime manages how, when and where each actor runs, and also routes messages between actors.

A large number of actors can execute simultaneously, and actors execute independently from each other.

## Actors in Dapr

Dapr includes a runtime that specifically implements the [Virtual Actor pattern](https://www.microsoft.com/research/project/orleans-virtual-actors/). With Dapr's implementation, you write your Dapr actors according to the actor model, and Dapr leverages the scalability and reliability guarantees that the underlying platform provides.

Every actor is defined as an instance of an actor type, identical to the way an object is an instance of a class. For example, there may be an actor type that implements the functionality of a calculator and there could be many actors of that type that are distributed on various nodes across a cluster. Each such actor is uniquely identified by an actor ID.

<img src="/images/actor_background_game_example.png" width=400>

## Dapr actors vs. Dapr Workflow

> Dapr actors and [Dapr Workflow]({{< ref workflow-overview.md >}}) are two different Dapr components that provide different features for building distributed applications.

Reviewer: ```suggestion
Dapr actors builds on the state management and service invocation APIs to create stateful, long running objects with identity. [Dapr Workflow]({{< ref workflow-overview.md >}}) and Dapr Actors are related, with workflows building on actors to provide a higher level of abstraction to orchestrate a set of actors, implementing common workflow patterns and managing the lifecycle of actors on your behalf.
```

Dapr actors are designed to provide a way to encapsulate state and behavior within a distributed system. An actor can be activated on demand by a client application. When an actor is activated, it is assigned a unique identity, which allows it to maintain its state across multiple invocations. This makes actors useful for building stateful, scalable, and fault-tolerant distributed applications.

> On the other hand, Dapr Workflow provides a way to define and orchestrate complex workflows that involve multiple services and components within a distributed system. Workflows allow you to define a sequence of steps or tasks that need to be executed in a specific order, and can be used to implement business processes, event-driven workflows, and other similar scenarios.

Reviewer: ```suggestion
On the other hand, Dapr Workflow provides a way to define and orchestrate complex workflows that involve multiple services and components within a distributed system. Workflows allow you to define a sequence of steps or tasks that need to be executed in a specific order, and can be used to implement business processes, event-driven workflows, and other similar scenarios.

As mentioned above, Dapr Workflow builds on Dapr Actors managing their activation and lifecycle.
```

Reviewer: I would consider adding a diagram here that shows the layering for statement management, actors and workflow. I an draw this for you and we can add this after the refactoring.

[Learn more about Dapr Workflow and how to use workflows in your application.]({{< ref workflow-overview.md >}})

## Features

### Actor lifetime

Since Dapr actors are virtual, they do not need to be explicitly created or destroyed. The Dapr actor runtime:
1. Automatically activates an actor once it receives an initial request for that actor ID. 
1. Garbage-collects the in-memory object of unused actors. 
1. Maintains knowledge of the actor's existence in case it's reactivated later.

An actor's state outlives the object's lifetime, as state is stored in the configured state provider for Dapr runtime.

[Learn more about actor lifetimes.]({{< ref "actors-features-concepts.md#actor-lifetime" >}})

### Distribution and failover

To provide scalability and reliability, actors instances are  throughout the cluster and Dapr distributes actor instances throughout the cluster and automatically migrates them to healthy nodes.

> #### Actor placement service

Reviewer: I would move this placement service into its own section since this level of detail is not needed in the overview. Still call this topic "Distribution and failover", just then say, read this new topic for more details. 

I would then make sure this topic "https://docs.dapr.io/concepts/dapr-services/placement/" links directly to this new topic in place of the "actors" currently there.

The Dapr actor runtime manages distribution scheme and key range settings for you via the actor `Placement` service. 

<img src="/images/actors_background_placement_service_registration.png" width=600>

When a new instance of a service is created:

1. The sidecar makes a call to the actor service to retrieve registered actor types and configuration settings.
1. The corresponding Dapr runtime registers the actor types it can create.
1. The `Placement` service calculates the partitioning across all the instances for a given actor type.

[Learn more about Dapr actor placement.]({{< ref "actors-features-concepts.md#actor-placement-service" >}})

#### Actor communication

> You can interact with Dapr to invoke the actor method by calling HTTP/gRPC endpoint, like the example in the diagram below.

Reviewer: ```suggestion
You can invoke actor methods by calling them over HTTP, as shown in the general example below.
```
